StreamChatClient_v3/WebSocketClient/WebSocketPingController.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class WebSocketPingController {
+    typealias PingAction = () -> Void
+    typealias ForceReconnectAction = () -> Void
+    
+    /// The time interval to ping connection to keep it alive.
+    static let pingTimeInterval: TimeInterval = 25
+    static let pongTimeoutTimeInterval: TimeInterval = 3
+    
+    private let timerType: Timer.Type
+    private let timerQueue: DispatchQueue
+    
+    /// The timer used for scheduling `ping` calls
+    private lazy var pingTimerControl: RepeatingTimerControl =
+        timerType.scheduleRepeating(timeInterval: Self.pingTimeInterval, queue: timerQueue) { [weak self] in self?.sendPing() }
+    
+    /// The pong timeout timer.
+    private var pongTimeoutTimer: TimerControl?
+    private let ping: () -> Void
+    private let forceReconnect: () -> Void
+    
+    /// Creates a ping controller.
+    /// - Parameters:
+    ///   - timerType: a timer type.
+    ///   - timerQueue: a timer dispatch queue.
+    ///   - ping: an action for `WebSocketClient` to send a ping.
+    ///   - forceReconnect: an action for `WebSocketClient` to force disconnect and reconnect.
+    init(timerType: Timer.Type,
+         timerQueue: DispatchQueue,
+         ping: @escaping PingAction,
+         forceReconnect: @escaping ForceReconnectAction) {
```

Comment:
You will make fun of me, but why aren't these callbacks a simple delegate with two methods, that `WebSocketClient` implements? I think the API will be much nicer.

Reply:
It's much more easy to use as I commented in our channel.